From: Niels Ole Salscheider <niels_ole@salscheider-online.de>
Date: Fri, 25 Mar 2016 14:36:10 +0000
Subject: Instantiate QApplication before calling sessionBus()
X-Git-Tag: v16.03.90
X-Git-Url: http://quickgit.kde.org/?p=ktp-text-ui.git&a=commitdiff&h=5015460c6fbd9696bf482e53e7cb94b64097db12
---
Instantiate QApplication before calling sessionBus()

Otherwise, the names can be available on the bus before the objects
are registered with QtDBus. This causes the text-ui to be empty when
started from ktp-contact-list with qt 5.6.

REVIEW: 127493
---


--- a/app/main.cpp
+++ b/app/main.cpp
@@ -51,6 +51,8 @@
     Tp::registerTypes();
     KTp::registerOtrTypes();
 
+    Tp::SharedPtr<TelepathyChatUi> app(new TelepathyChatUi(argc, argv));
+
     Tp::ChannelFactoryPtr channelFactory = Tp::ChannelFactory::create(QDBusConnection::sessionBus());
     channelFactory->addCommonFeatures(Tp::Channel::FeatureCore);
 
@@ -64,7 +66,6 @@
 
     Tp::ClientRegistrarPtr registrar = Tp::ClientRegistrar::create(KTp::accountFactory(), KTp::connectionFactory(), channelFactory, KTp::contactFactory());
 
-    Tp::SharedPtr<TelepathyChatUi> app(new TelepathyChatUi(argc, argv));
     Tp::AbstractClientPtr handler = Tp::AbstractClientPtr(app);
     registrar->registerClient(handler, QLatin1String(KTP_TEXTUI_CLIENT_NAME));
 

